src/formatDate.js

```diff
@@ -8,7 +8,51 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(/[-/.]/);
+  const valueMap = fromFormat.reduce((accumulator, format, index) => {
+    accumulator[format] = dateParts[index];
+
+    return accumulator;
+  }, {});
+
+  const yy = Number(valueMap['YY']);
+  const year =
+    valueMap['YYYY'] ||
+    (yy >= 0 && yy <= 99
+      ? yy < 30
+        ? `20${String(yy).padStart(2, '0')}`
+        : `19${String(yy).padStart(2, '0')}`
+      : valueMap['YY']);
+
+  const month = valueMap['MM'] ? Number(valueMap['MM'], 10) - 1 : 0;
+  const day = valueMap['DD'] ? Number(valueMap['DD'], 10) : 1;
+
+  if (!year) {
+    return null;
+  }
+
+  const parsedDate = new Date(year, month, day);
+
+  if (isNaN(parsedDate.getTime())) {
+    return null;
+  }
+
+  const formattedDate = toFormat.map((formatPart) => {
+    switch (formatPart) {
+      case 'YYYY':
+        return parsedDate.getFullYear();
+      case 'YY':
+        return parsedDate.getFullYear().toString().slice(-2);
+      case 'MM':
+        return (parsedDate.getMonth() + 1).toString().padStart(2, '0');
+      case 'DD':
+        return parsedDate.getDate().toString().padStart(2, '0');
+      default:
+        return formatPart;
+    }
+  });
+
+  return formattedDate.join(toFormat[3]).slice(0, -2);
```

Comment:
The slicing of the return string is not necessary and can lead to incorrect results if the `toFormat` separator is not a single character or if the `toFormat` pattern ends with a date part instead of a separator. You should join the formatted date parts using the last element of the `toFormat` array without slicing.